### DISCUSSION TITLE
BM23 stuttering audio issue

### DISCUSSION THREAD

**Discussion by audioFail1** [tomaskovacik/IS2020#6] [General]
Hello,

I am having an issue with the BM23  Bluetooth module (IS2023). I saw that you have laid out a breakout board for the BM23, so I am guessing you have some experience with it. 

The Issue:
Audio from the BM23 pauses roughly every 2 seconds for about 300ms. I know the source of the issue is the BM23 and not some other part of the circuit (such as the amplifier). Because this pause occurs on the DT0 line from the BM23 to the amplifier. This was measured on an oscilloscope. 

What I Have Tried:
I have successfully written to the BM23 using the UI, DSP, MPET and EEPROM tools. I have changed all of the configuration settings that I thought may have some impact on this issue (and some that I thought wouldn’t but tried anyway) in an attempt to solve this issue. I have also configured the circuit similar to the reference circuit (provided in the BM23 datasheet), thinking that some noise on one of the pins maybe causing this issue (I will provide a schematic of my test circuit).

I have tried several A2DP sources (several phones and a tablet), with no change in the issue.

I tried a second BM23 chip, thinking it could have been a soldering, overheating or ESD issue. But the second chip performs the same as the first.

I can provide screen shots of my configuration settings for both the UI and DSP tools, if needed.

Any help would be greatly appreciated.

Below is a schematic of my current test circuit using the BM23.

[Speaker1t.PDF](https://github.com/tomaskovacik/IS2020/files/6612349/Speaker1t.PDF)

**Comment by tomaskovacik** [comment]
I have no problem with my bm23
https://github.com/tomaskovacik/kicad-library/blob/master/library/datasheet/BM23/notes.md 

my config: https://github.com/tomaskovacik/kicad-library/tree/master/library/datasheet/BM23

**Reply by audioFail1** [reply]
Hey Tomas,

Thanks for the reply!

Your .ipf file does work, the issue I described is gone when I use your .ipf file. I tried merging your UI tool settings with my DSP tool settings and the problem was still present. So, I am assuming the setting that affects the issue I am having is within the DSP settings.

Questions:
1) Are you familiar with the issue I have been describing?

2) Could you provide screen shots of your DSP settings?

3) Did you modify the .ipf file using some other means than the UI, DSP and MPET tools?

Thanks for any help!

Tony

**Reply by tomaskovacik** [reply]
I did not use any custom dsp setting

in notes.md  are all files used in my setup

all files are from IS2021_203_tools.zip from microchip, also inside this zip archive is pdf IS2023_203 App Note.pdf which mentioned some patch named BM23_203_AFHIJKLMP.txt and dsp file: BM23_203_DSP_new.txt

**Reply by audioFail1** [reply]
Hey Tomas,

Thanks again for the reply!

That was the issue, I was not including the BM23_203_AFHIJKLMP.txt file in my "merge" list. I guess I should just follow the instructions, right? I was in the app. notes, which I read but clearly missed that step.

Thanks very much for your help.

Tony

**Reply by tomaskovacik** [reply]
no problem, my module restarted each time I played something first time I played with it, I just do not remembered it :)
